homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,171 @@
+
+            errors["base"] = "no_selection"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONST_REGION_A_TO_D): cv.multi_select(
+                        self.regions[CONST_REGION_A_TO_D]
+                    ),
+                    vol.Optional(CONST_REGION_E_TO_H): cv.multi_select(
+                        self.regions[CONST_REGION_E_TO_H]
+                    ),
+                    vol.Optional(CONST_REGION_I_TO_L): cv.multi_select(
+                        self.regions[CONST_REGION_I_TO_L]
+                    ),
+                    vol.Optional(CONST_REGION_M_TO_Q): cv.multi_select(
+                        self.regions[CONST_REGION_M_TO_Q]
+                    ),
+                    vol.Optional(CONST_REGION_R_TO_U): cv.multi_select(
+                        self.regions[CONST_REGION_R_TO_U]
+                    ),
+                    vol.Optional(CONST_REGION_V_TO_UE): cv.multi_select(
+                        self.regions[CONST_REGION_V_TO_UE]
+                    ),
+                    vol.Required(CONF_MESSAGE_SLOTS, default=5): vol.All(
+                        int, vol.Range(min=1, max=20)
+                    ),
+                    vol.Required(CONF_FILTER_CORONA, default=True): cv.boolean,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    def swap_key_value(dict_to_sort: dict[str, str]) -> dict[str, str]:
+        """Swap keys and values in dict."""
+        all_region_codes_swaped: dict[str, str] = {}
+
+        for key, value in dict_to_sort.items():
+            if value not in all_region_codes_swaped:
+                all_region_codes_swaped[value] = key
+            else:
+                for i in range(len(dict_to_sort)):
+                    tmp_value: str = value + "_" + str(i)
+                    if tmp_value not in all_region_codes_swaped:
+                        all_region_codes_swaped[tmp_value] = key
+                        break
+
+        return dict(sorted(all_region_codes_swaped.items(), key=lambda ele: ele[1]))
+
+    def split_regions(self) -> None:
+        """Split regions alphabetical."""
+        for index, name in self._all_region_codes_sorted.items():
+            if name[0] in CONST_LIST_A_TO_D:
+                self.regions[CONST_REGION_A_TO_D][index] = name
+            if name[0] in CONST_LIST_E_TO_H:
+                self.regions[CONST_REGION_E_TO_H][index] = name
+            if name[0] in CONST_LIST_I_TO_L:
+                self.regions[CONST_REGION_I_TO_L][index] = name
+            if name[0] in CONST_LIST_M_TO_Q:
+                self.regions[CONST_REGION_M_TO_Q][index] = name
+            if name[0] in CONST_LIST_R_TO_U:
+                self.regions[CONST_REGION_R_TO_U][index] = name
+            if name[0] in CONST_LIST_V_TO_UE:
+                self.regions[CONST_REGION_V_TO_UE][index] = name
```

Comment:
Consider using something like:

```suggestion
        for index, name in self._all_region_codes_sorted.items():
           for region_name, grouping_letters in CONST_REGION_MAPPING.items():
              if name[0] in grouping_letters:
                 self.regions[region_name][index] = name
                 break
```